becquerel/core/spectrum.py

```diff
@@ -806,6 +807,55 @@ def combine_bins(self, f):
         obj = Spectrum(**kwargs)
         return obj
 
+    def rebin(self, out_edges, method="interpolation", slopes=None,
+              zero_pad_warnings=True):
+        """
+        Spectra rebinning via deterministic or stochastic methods.
+
+        Args:
+            out_edges (np.ndarray): an array of the output bin edges
+                [num_channels_out]
+            method (str): rebinning method
+                "interpolation"
+                    Deterministic interpolation
+                "listmode"
+                    Stochastic rebinning via conversion to listmode of energies
+            slopes (np.ndarray|None): (optional) an array of input bin slopes
+                for quadratic interpolation
+                (only applies for "interpolation" method)
+                [len(spectrum) + 1]
+        zero_pad_warnings (boolean): warn when edge overlap results in
+            appending empty bins
+
+        Raises:
+            SpectrumError: for bad input arguments
+
+        Returns:
+            A new Spectrum object with the rebinned data.
+        """
+        if self.bin_edges_kev is None:
+            raise SpectrumError('Cannot rebin spectrum without energy '
+                                'calibration')
+        if ((method.lower() == 'listmode') and
+                (self._counts is None) and
+                (self.livetime is not None)):
```

Comment:
Not sure if the last check is needed, maybe better to leave it out and have the warning here. If livetime is not defined it will fail further down (at `self.counts_vals`) anyways and the warning helps to see what is going on here.